src/search-headless.ts

```diff
@@ -411,6 +411,9 @@ export default class SearchHeadless {
     const { limit, offset, sortBys, locationRadius } = this.state.vertical;
     const { referrerPageUrl, context } = this.state.meta;
     const { userLocation } = this.state.location;
+    const zeroOffset = this.state.vertical.offset ?? 0;
+    const inputIsPreviousQuery = input == this.state.query.mostRecentSearch;
+    const queryId = !zeroOffset || inputIsPreviousQuery ? this.state.query.queryId : undefined;
```

Comment:
`(!zeroOffset || inputIsPreviousQuery) ? queryId : undefined` to make it clearer?

Reply:
sure thing